src/cli/cli_coap_secure.cpp

```diff
@@ -366,7 +366,7 @@ void CoapsSecure::HandleServerResponse(otCoapHeader *aHeader, otMessage *aMessag
 {
     otError      error = OT_ERROR_NONE;
     otCoapHeader responseHeader;
-    otMessage *  responseMessage;
+    otMessage *  responseMessage   = NULL;
```

Comment:
Static analysis should catch this kind of issue. What code path does the lack of initialization here cause an issue?

Reply:
No, I don't think there is such a code path. But some compilers do complain this:
```
  CXX      libopenthread_cli_ftd_a-cli_coap_secure.o
cli_coap_secure.cpp: In member function 'void ot::Cli::CoapsSecure::HandleServerResponse(otCoapHeader*, otMessage*, const otMessageInfo*)':
cli_coap_secure.cpp:442:22: error: 'responseMessage' may be used uninitialized in this function [-Werror=maybe-uninitialized]
         otMessageFree(responseMessage);
         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```